pages/_app.tsx

```diff
@@ -1,8 +1,38 @@
-import { AppProps } from 'next/app'
-import '../styles/index.css'
+import { AppProps } from "next/app";
+import "../styles/index.css";
+import Router from "next/router";
+import { PageLoader } from "../components/PageLoader";
+
+import { AnimatePresence } from "framer-motion";
+import { useEffect, useState } from "react";
 
 function MyApp({ Component, pageProps }: AppProps) {
-  return <Component {...pageProps} />
+  const [loading, setLoading] = useState(false);
```

Comment:
loading by default should be true as it will flicker otherwise

Reply:
Indeed, the loading animation triggers upon route changes. If we were to set it to true initially, the loading screen would consistently display, even when the route isn't changing.